### ISSUE TITLE
append_fields in template leads to invalid template if parent key is already contained

### ISSUE THREAD

**Issue by simitt** [elastic/apm-server#1959]
In case additional fields should be added to the ES index template, one can use the experimental setting `setup.template.append_fields` in the `apm-server.yml`.
The setting cannot be used to change the type of an existing field, but to add new fields. If the field is already contained in the mapping an error is raised. 

As @axw reported adding `context.request.socket.remote_address` raises an error, although the field is not contained in the template. Some investigations suggest that there is a bug in the implementation using [`field.HasNode`](https://github.com/elastic/beats/blob/master/libbeat/common/field.go#L174), which does not allow to add additional keys under an existing leaf, even if it is of type `object`.

**Comment by simitt** [comment]
There's a PR up in libbeat https://github.com/elastic/beats/pull/10959, that would need to be backported to >= 6.6.

The bug currently only becomes obvious in `6.6`, as we added non indexed fields to the fields.yml file for documentation purposes. As soon as those fields are part of the template, the current implementation of `setup.template.append_fields` does not allow to add further keys under a leaf node, even when it is of type `object`. The PR in beats aims to fix this. 

From > 6.6 on we removed the non indexed fields from the template again (when reverting the aliases). @graphaelli I am not sure whether or not the bugfix in libbeat is going to be backport to 6.6. I suggest that depending on the backports we either add the non indexed fields also from > 6.6 on or we remove them again from 6.6 in the next patch.
